pytest_perf/plugin.py

```diff
@@ -36,9 +62,18 @@ def pytest_sessionfinish():
 
 
 class File(pytest.File):
+    def __init__(self, fspath, parent, target=".", baseline=None):
+        super().__init__(fspath, parent)
+        self.target = target
+        self.baseline = baseline
+
     def collect(self):
         return (
-            Experiment.from_parent(self, name=f'{self.name}:{spec["name"]}', spec=spec)
+            Experiment.from_parent(
+                self,
+                name=f'{self.name}:{spec["name"]}',
+                spec={"target": self.target, "baseline": self.baseline, **spec},
```

Comment:
I'd like to avoid altering the spec here. Spec is meant to be the specification of the test.